src/components/Stepper/Step/Step.tsx

```diff
@@ -0,0 +1,118 @@
+import classNames from "classnames";
+import React from "react";
+import Icon from "components/Icon";
+import Link, { LinkProps } from "components/Link";
+import { ClassName } from "types";
+import "./Step.scss";
+
+export type Props = {
+  /**
+   * Whether the step has a darkened progress line.
+   */
+  hasProgressLine: boolean;
+  /**
+   * Index of the step.
+   */
+  index: number;
+  /**
+   * Title of the step.
+   */
+  title: string;
+  /**
+   * Optional label for the step.
+   */
+  label?: string;
+  /**
+   * Optional props to configure the `Link` component.
+   */
+  linkProps?: LinkProps;
+  /**
+   * Whether the step is clickable. If set to false, the step is not clickable and the text is muted with a light-dark colour.
+   */
+  enabled: boolean;
+  /**
+   * Optional value to highlight the selected step.
+   */
+  selected?: boolean;
+  /**
+   * Icon to display in the step. Specify "number" if the index should be displayed.
+   */
+  iconName: string;
```

Comment:
Should we reference the icon names `ValueOf<typeof ICONS> | string | "number"` here to allow autocomplete?

Reply:
While having autocomplete would be nice, adding the general `string` type to the union expands the type, which prevents ts from providing autocomplete suggestions.